airflow/providers/cncf/kubernetes/operators/kubernetes_pod.py

```diff
@@ -484,6 +489,11 @@ def create_pod_request_obj(self) -> k8s.V1Pod:
             ),
         )
 
+        if self.random_name_suffix:
+            if pod_template:
+                pod_template.metadata.name = PodGenerator.make_unique_pod_id(pod_template.metadata.name)
+            pod.metadata.name = PodGenerator.make_unique_pod_id(pod.metadata.name)
+
         pod = PodGenerator.reconcile_pods(pod_template, pod)
```

Comment:
```suggestion
        pod = PodGenerator.reconcile_pods(pod_template, pod)
        
        if self.random_name_suffix:
            pod.metadata.name = PodGenerator.make_unique_pod_id(pod.metadata.name)
```

Can we randomize after we reconcile the pod?

Reply:
Sure, I'll move this.  Is there a particular reason that I missed or just preference?

Reply:
See: https://github.com/apache/airflow/pull/19398/commits/2d34a160b89fa3217cce9ce7996a4787660b799b